processor/samplingprocessor/processor.go

```diff
@@ -111,7 +111,13 @@ func (sp *tracesSamplingProcessor) processTraces(ctx context.Context, td ptrace.
 				return err == nil && match && sampleFunc(sp.dropCutOffRatio)
 			})
 		}
+		td.ResourceSpans().At(i).ScopeSpans().RemoveIf(func(ss ptrace.ScopeSpans) bool {
+			return ss.Spans().Len() == 0
+		})
```

Comment:
Could we add some go test that verifies this new behavior?

Reply:
Added some tests, had to use `require.NotEqual(t, 0, logRecords)` instead of NotEmpty. For whatever reason it couldn't detect an empty pmetric slice. I verified the tests expose the old behavior by commenting out the fixes